runtime/src/constants.rs

```diff
@@ -152,59 +149,83 @@ lazy_static! {
 
 // Change it when changing the currency constants!
 parameter_types! {
-    pub const ExistentialDeposit: u128 = 1;
+    pub const ExistentialDeposit: Balance = 1;
 }
 
 pub mod currency {
     use super::Balance;
 
-    pub const JOYS: Balance = 250_000_000;
-    pub const DOLLARS: Balance = JOYS / 12500; // 20_000
-    pub const CENTS: Balance = DOLLARS / 100; // 200
-
+    /// The base currency unit.
+    pub const HAPI: Balance = 1;
+    /// One JOY equals 10 Billion HAPIs. Hence we use 10 decimal places in
+    /// currency representation.
+    pub const JOY: Balance = 10_000_000_000 * HAPI;
+    /// The planned initial token issuance of JOY at genesis.
+    const TOTAL_JOY_ISSUANCE_AT_GENESIS: Balance = 1_000_000_000 * JOY;
+    /// A Hypothetical valuation of total issued JOY tokens in USD.
+    const ESTIMATED_JOY_USD_MCAP: Balance = 100_000_000;
+    // Constants used to derive balance configurations of pallets more human readable
+
+    /// Balance estimated worth one USD.
+    pub const DOLLARS: Balance = TOTAL_JOY_ISSUANCE_AT_GENESIS / ESTIMATED_JOY_USD_MCAP;
```

Comment:
- Is this division going to be safe/functional as people start varying those other variables? in partiuclar the mcap?
- Can we find a more direct name? like `HAPIs_IN_A_DOLLAR` or something?

Reply:
.. But I see the benefit in keeping, as long as the the MCAP is "regulary" updated to match real world value, it will keep fees and stakes and bonds more in tune with real world values.

Reply:
used saturating_div and saturating_mul for safety